### PR TITLE
[PLUGIN-1721] Added support for timestamp_tz datatype in. snowflake source plugin.

### PR DESCRIPTION
**Pull request by vikasrathee-cs**
Added support for timestamp_tz datatype in snowflake plugin.

### CODE REVIEW

File: docs/Snowflake-batchsource.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ Data Types Mapping
 | TIMESTAMP                      | timestamp/string      | Alias for one of the TIMESTAMP variations (TIMESTAMP_NTZ by default).|
 | TIMESTAMP_LTZ                  | timestamp             | TIMESTAMP with local time zone; time zone, if provided, is not stored.|
 | TIMESTAMP_NTZ                  | timestamp             | TIMESTAMP with no time zone; time zone, if provided, is not stored.|
-| TIMESTAMP_TZ                   | string                | TIMESTAMP with time zone.                                 |
+| TIMESTAMP_TZ                   | timestamp             | TIMESTAMP with time zone.                                 |
```
Comment (albertshau) [inline docs/Snowflake-batchsource.md:93]: was this mistakenly in the doc before or was it actually being read as a string?

If it was being read as a string, we can't just change the data type as that will break backwards compatibility.

Comment (vikasrathee-cs) [reply docs/Snowflake-batchsource.md:93]: it was added mistakenly only, was giving validation error on this data type.

File: src/test/java/io/cdap/plugin/snowflake/common/util/SchemaHelperTest.java [hunk 2]
```diff
@@ -137,7 +138,8 @@ public void testGetSchemaFromSnowflake() throws IOException {
       Schema.Field.of("field121", Schema.nullableOf(Schema.decimalOf(38))),
       Schema.Field.of("field122", Schema.decimalOf(38)),
       Schema.Field.of("field131", Schema.nullableOf(Schema.decimalOf(38))),
-      Schema.Field.of("field132", Schema.decimalOf(38))
+      Schema.Field.of("field132", Schema.decimalOf(38)),
+      Schema.Field.of("field133", Schema.of(Schema.LogicalType.TIMESTAMP_MICROS))
```
Comment (MrRahulSharma) [inline src/test/java/io/cdap/plugin/snowflake/common/util/SchemaHelperTest.java]: Also add a field which is nullable of Timestamp_micros as well.

File: src/test/java/io/cdap/plugin/snowflake/common/util/SchemaHelperTest.java [hunk 3]
```diff
@@ -110,7 +110,8 @@ public void testGetSchemaFromSnowflake() throws IOException {
       new SnowflakeFieldDescriptor("field121", Types.BIGINT, true),
       new SnowflakeFieldDescriptor("field122", Types.BIGINT, false),
       new SnowflakeFieldDescriptor("field131", Types.SMALLINT, true),
-      new SnowflakeFieldDescriptor("field132", Types.SMALLINT, false));
+      new SnowflakeFieldDescriptor("field132", Types.SMALLINT, false),
+      new SnowflakeFieldDescriptor("field133", Types.TIMESTAMP_WITH_TIMEZONE, false));
```
Comment (MrRahulSharma) [inline src/test/java/io/cdap/plugin/snowflake/common/util/SchemaHelperTest.java]: Also add a field which is nullable of Timestamp_micros as well.

File: src/test/java/io/cdap/plugin/snowflake/common/util/SchemaHelperTest.java (hunk 2 above)
Comment (vikasrathee-cs) [reply src/test/java/io/cdap/plugin/snowflake/common/util/SchemaHelperTest.java]: added

File: docs/Snowflake-batchsource.md [hunk 4]
```diff
@@ -90,7 +90,7 @@ Data Types Mapping
 | TIMESTAMP                      | timestamp/string      | Alias for one of the TIMESTAMP variations (TIMESTAMP_NTZ by default).|
```
Comment (MrRahulSharma) [inline docs/Snowflake-batchsource.md]:
```suggestion
| TIMESTAMP                      | timestamp      | Alias for one of the TIMESTAMP variations (TIMESTAMP_NTZ by default).|
```

**Review by MrRahulSharma — APPROVED** [review]

**Pull request merged**